inbm/dispatcher-agent/dispatcher/fota/bios_factory.py

```diff
@@ -195,6 +194,20 @@ def __init__(self,
                  params: Dict) -> None:
         super().__init__(dispatcher_broker, repo, params)
 
+    def _get_guid_from_system(self, manifest_guid: Optional[str]) -> str:
+        # get the GUID from the system using FW tool
+        extracted_guids = extract_guids(
+            self._fw_tool, ["System Firmware type", "system-firmware type"])
+        if manifest_guid:
+            if manifest_guid not in extracted_guids:
+                raise FotaError(
+                    f"GUID in manifest does not match any system firmware GUID on the system")
+            else:
+                return manifest_guid
+        else:
+            return extracted_guids[0]
```

Comment:
<h3>$\color{#FFA500}Improvements: Code Robustness$</h3>

Consider adding error handling for the case where `extracted_guids` is empty before attempting to access its first element. This could prevent potential `IndexError` exceptions if no GUIDs are found.